kaldb/src/main/java/com/slack/kaldb/metadata/core/EphemeralMutableMetadataStore.java

```diff
@@ -35,6 +39,11 @@ public EphemeralMutableMetadataStore(
     }
   }
 
+  public Object createSync(T metadataNode)
+      throws ExecutionException, InterruptedException, TimeoutException {
```

Comment:
What if these returned an `Optional<T>?` That would allow us to avoid casting the returned object and individually handling these exceptions.